### PR TITLE
Remove s3 WSClient. Replace with S3 API

### PR DESCRIPTION
**Pull request by QuarpT**
## What does this change?

Removes the custom S3 WSClient - it's only used in one place and seems unnecessary. The difference is that WSClient is non-blocking - so I've wrapped the amazon API in blocking operations thread pool. 

If we want to avoid using the native Java Amazon API, then we should use https://developer.lightbend.com/docs/alpakka/current/s3.html rather than implement our own http client.

## What is the value of this and can you measure success?

Simplifies retrieving facia fronts

## Tested in CODE?

I will

### CODE REVIEW

File: facia/app/controllers/front/FrontJsonFapi.scala [hunk 1]
```diff
@@ -1,29 +1,41 @@
...
 import model.PressedPage
-import services.PressedPageService
+import play.api.libs.json.Json
+import services.S3
 import scala.concurrent.{ExecutionContext, Future}
 
 trait FrontJsonFapi extends Logging {
   lazy val stage: String = Configuration.facia.stage.toUpperCase
   val bucketLocation: String
+  val parallelJsonPresses = 16
```
Comment (QuarpT) [inline facia/app/controllers/front/FrontJsonFapi.scala]: Increasing to 16 after seeing sporadic exceptions in prod with a low threshold of 8.

File: facia/app/controllers/front/FrontJsonFapi.scala [hunk 2]
```diff
@@ -1,29 +1,41 @@
...
+  def blockingOperations: BlockingOperations
 
   private def getAddressForPath(path: String, format: String): String = s"$bucketLocation/${path.replaceAll("""\+""", "%2B")}/fapi/pressed.v2.$format"
 
   def get(path: String)(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] = errorLoggingF(s"FrontJsonFapi.get $path") {
-    pressedPageService.get(getAddressForPath(path, "json"))
+    pressedPageFromS3(getAddressForPath(path, "json"))
+  }
+
+  private def pressedPageFromS3(path: String)(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] = errorLoggingF(s"FrontJsonFapi.pressedPageFromS3 $path") {
```
Comment (TBonnin) [inline facia/app/controllers/front/FrontJsonFapi.scala]: Is this private methode necessary? Can its content be moved into `get` above?

File: facia/app/services/PressedPageService.scala [hunk 3]
```diff
@@ -1,55 +0,0 @@
...
-        _.flatMap { json =>
-          val stopWatch: StopWatch = new StopWatch
-          val pressedPage = Json.fromJson[PressedPage](json) match {
-            case JsSuccess(page, _) =>
-              Option(page)
-            case JsError(errors) =>
-              log.warn("Could not parse JSON in FrontJson")
-              None
-          }
-          FaciaPressMetrics.FrontDecodingLatency.recordDuration(stopWatch.elapsed)
```
Comment (TBonnin) [inline facia/app/services/PressedPageService.scala]: Can we keep this metric recording?

**Review by TBonnin — CHANGES_REQUESTED** [review]

File: facia/app/services/PressedPageService.scala (hunk 3 above)
Comment (QuarpT) [reply facia/app/services/PressedPageService.scala]: I will put it back

File: facia/app/controllers/front/FrontJsonFapi.scala (hunk 2 above)
Comment (QuarpT) [reply facia/app/controllers/front/FrontJsonFapi.scala]: It can be

**Review by TBonnin — APPROVED** [review]

**Pull request merged**